### ISSUE TITLE
/System/Library/Extensions/Proxmark3.kext/Contents: Read-only file system

### ISSUE THREAD

**Issue by echo-cool** [Proxmark/proxmark3#922]
While I try to use `sudo make install_kext`
There is an error.

MacOS 10.15.3

```
mkdir -p /System/Library/Extensions/Proxmark3.kext/Contents
mkdir: /System/Library/Extensions/Proxmark3.kext/Contents: Read-only file system
make: *** [Makefile:39: install_kext] Error 1
```

![image](https://user-images.githubusercontent.com/25057326/77021687-665a7e80-69c2-11ea-9bb4-43baeeb896e8.png)

**Comment by pwpiwi** [comment]
You are walking on ancient grounds :smiley:. I don't own or know much about MacOS. If this compares to Linux then probably the respective filesystem needs to be remounted read/write before `make install_kext`. However, I would assume that there is a more standard way to add Extensions...

**Comment by pwpiwi** [comment]
Please try Pull Request #923 which should do the remount when running 'sudo make kext_install'

**Comment by iceman1001** [comment]
...like Piwi said,   why are you using the **hid-flasher** ?  We are talking pre-2013 now.

**Comment by echo-cool** [comment]
while, the firmware of my proxmark3 is quite old.... I just follow the instructions to upgrade....
![image](https://user-images.githubusercontent.com/25057326/77047533-d804ed80-69ff-11ea-91b8-8484f05838d3.png)

> Please try Pull Request #923 which should do the remount when running 'sudo make kext_install'
Thanks, Let me try.

**Comment by echo-cool** [comment]
> ...like Piwi said, why are you using the **hid-flasher** ? We are talking pre-2013 now.

It seems like it still can't work. Maybe to make `/system` unchangeable is the new feature of Catalina to protect the MacOS ?
![image](https://user-images.githubusercontent.com/25057326/77048370-472f1180-6a01-11ea-8137-cc536f31c25d.png)

**Comment by echo-cool** [comment]
oh, I solved this problem by switching system.
Thanks

**Comment by pwpiwi** [comment]
Switching the system is not really a fix. 😃 

I have added the root directory to the mount command. If you don't mind, can you please `sudo make install_kext` again. Some users may benefit in the future.

**Comment by alfonsrv** [comment]
Still does not work, same error. Upgrade wiki for macOS seems outdated.

**Comment by pwpiwi** [comment]
You are obviously using the RRG repository and therefore should report issues there: https://github.com/RfidResearchGroup/proxmark3/issues

**Comment by alfonsrv** [comment]
No. I tried using the RRG repository before and – after I got the error my board would not support the commands yet, I thought I would have to update the bootloader manually before I could use the more "automatic" update process.  
So I resorted to the Proxmark/proxmark3 repo and followed the macOS steps, which puts me in the position to confirm the kext issue is still very much present under `macOS 10.15.5`. Ignoring the HID flasher and the connected kext, instead using the `client/flasher` worked though.

Afterwards I ran into an issue where the `proxmark3` cli client would no longer talk to my Proxmark, before I read somewhere that the clients are not intercompatible with the firmware anymore. So this issue was ultimately resolved by using the RRG repository instead and manually flashing their image instead.  
Besides the discrepancies with macOS Catalina clearly not being supported properly in the process outlined in the wiki, I must say it is somewhat confusing to have no note anywhere how much RRG and this repo have formed a defacto two-world system, other than on some nested sub-pages and forum posts.

**Comment by iceman1001** [comment]
....
if you get that message.... 
you should flash bootrom first like the instructions says.

```
./pm3-flash-bootrom
./pm3-flash-fullimage
```

Its the _normal_ way to switch to RRG/Iceman repo.

**Comment by pwpiwi** [comment]
> which puts me in the position to confirm the kext issue is still very much present under macOS 10.15.5

Did you apply the changes in https://github.com/Proxmark/proxmark3/pull/923/files before running `sudo make install_kext`?

**Comment by alfonsrv** [comment]
> Did you apply the changes in https://github.com/Proxmark/proxmark3/pull/923/files before running `sudo make install_kext`?

Sorry, thought it was already merged. I did not.

**Comment by pwpiwi** [comment]
No it isn't merged. I don't have access to a MacOS machine and I don't like to merge untested changes.

**Comment by alfonsrv** [comment]
Probably a setup Virtualbox image out there. But I'll try the code on a lab Hackintosh on Wednesday and will report back.

**Comment by mattop4d** [comment]
Came here to confirm that the fix in the PR is not working. I got this error:

```
$ sudo make install_kext

spctl --master-disable
mount -uw /
mount_apfs: volume could not be mounted: Operation not permitted
mount: / failed with 77
make: *** [install_kext] Error 77
```

Tested on a Macbook Pro with MacOS Catalina 10.15.6

**Comment by pwpiwi** [comment]
See https://www.tonymacx86.com/threads/macos-10-15-catalina-how-to-mount-root-partition-read-write-for-post-installation.283749/page-2#post-2018467 for some hint
